pkg/virt-api/api.go

```diff
@@ -1033,19 +1033,23 @@ func (app *virtAPIApp) Run() {
 
 	// Run informers for webhooks usage
 	webhookInformers := webhooks.GetInformers()
+	kubeInformerFactory := controller.NewKubeInformerFactory(app.virtCli.RestClient(), app.virtCli, app.namespace)
+	configMapInformer := kubeInformerFactory.ConfigMap()
 
 	stopChan := make(chan struct{}, 1)
 	defer close(stopChan)
 	go webhookInformers.VMIInformer.Run(stopChan)
 	go webhookInformers.VMIPresetInformer.Run(stopChan)
 	go webhookInformers.NamespaceLimitsInformer.Run(stopChan)
-	go webhookInformers.ConfigMapInformer.Run(stopChan)
+	go configMapInformer.Run(stopChan)
 
 	cache.WaitForCacheSync(stopChan,
 		webhookInformers.VMIInformer.HasSynced,
 		webhookInformers.VMIPresetInformer.HasSynced,
 		webhookInformers.NamespaceLimitsInformer.HasSynced,
-		webhookInformers.ConfigMapInformer.HasSynced)
+		configMapInformer.HasSynced)
+
+	app.clusterConfig = virtconfig.NewClusterConfig(configMapInformer.GetStore(), app.namespace)
```

Comment:
why wasn't the previous informer used like this?

```app.clusterConfig = virtconfig.NewClusterConfig(webhookInformers.ConfigMapInformer.GetStore(), app.namespace)```

Reply:
> hey. I'm a little confused about what's going on here with #2203 and this PR. I understand the code, and generally agree, but I'm unsure about how you want to proceed with these PRs.
> 
> Are you wanting pr #2203 merged, and then to rebase this PR in order to reduce it down to just the feature-gate changes? Otherwise it looks like this PR is a superset of both PRs.

Hi :) exactly - some of the feature-gates related changes depend on those in #2203 (the ones within the webhooks) so I branched from #2203 and continued working on this, assuming this PR will get ready after #2203 is merged and I'll then rebase it

Reply:
First, I think we will need a singleton for the cluster config, second I would recommend that we keep the informer. We will need at the end a controller around ClusterConfig, where we can register callbacks. They can then do things in our components, like restarting the whole component, restarting controllers, and so on.

Reply:
> First, I think we will need a singleton for the cluster config,

Yeah, that would simplify the wiring that makes the code a bit cumbersome now. Not sure about changing the ClusterConfig to Singleton though, maybe when [wire](https://github.com/google/wire) gets mature it could provide easier DI for us? I would prefer to think about this separately though - we anyway want the feature gates to be part of ClusterConfig I think.

> second I would recommend that we keep the informer inside the ClusterConfig. We will need at the end a controller around ClusterConfig, where we can register callbacks. They can then do things in our components, like restarting the whole component, restarting controllers, and so on.

Right, it makes a lot of sense for ClusterConfig to evolve that way. Done - the informer is now stored inside ClusterConfig